packages/explain-plan-helper/src/get-execution-stats.ts

```diff
@@ -109,12 +112,16 @@ function getIndexesFromStages(
     .map((index: string) => ({ index, shard: shard ?? null, fields: {} }));
 }
 
-function getExecutionTime(stats: ExecutionStats, stages: Stage[]): number {
-  // We sum executionTimeMillisEstimate from all the stages, except for first
-  // as we use its executationStats.executionTimeMillis
-  const [, ...allStages] = stages;
+function getAggregationExecutionTime(
+  stats: ExecutionStats,
+  stages: Stage[]
+): number {
   return (
-    (stats.executionTimeMillis ?? 0) +
-    sumArrayProp(allStages, 'executionTimeMillisEstimate')
+    // Aggregation ecxecution time is either accessible as part of stats, or as
+    // the estimated time of the last stage as execution time for stages is
+    // accumulated: every next stage includes the time for the previous stages
```

Comment:
I think in the UI it is already intuitive, to the point that this bug ticket was opened because we were doing something users weren't expecting us to do. I added this comment mostly for us so that we don't change this accidentally in the future. Does this makes sense?

Reply:
Sounds good - yup - thanks!